pkg/controller/config/config.go

```diff
@@ -0,0 +1,236 @@
+				}
+			}
+			if !found {
+				errs = append(errs, fmt.Errorf("invalid serviceCIDR option: %s, available values are: %s", serviceCIDR, clusterConfig.Spec.ServiceNetwork))
+			}
+		} else {
+			errs = append(errs, fmt.Errorf("serviceCIDR option can not be empty"))
+		}
+	}
+	if operConfig.Spec.AntreaImage == "" {
+		errs = append(errs, fmt.Errorf("antreaImage option can not be empty"))
+	}
+	if len(errs) > 0 {
+		return fmt.Errorf("invalidate configuration: %v", errs)
+	}
+	return nil
+}
+
+func NeedApplyChange(preConfig, curConfig *operatorv1.AntreaInstall) (bool, bool) {
+	if preConfig == nil {
+		return true, true
+	}
+
+	agentNeedChange := false
+	controllerNeedChange := false
+	if preConfig.Spec.AntreaAgentConfig != curConfig.Spec.AntreaAgentConfig {
+		agentNeedChange = true
+	}
+	if preConfig.Spec.AntreaCNIConfig != curConfig.Spec.AntreaCNIConfig {
+		agentNeedChange = true
+	}
+	if preConfig.Spec.AntreaControllerConfig != curConfig.Spec.AntreaControllerConfig {
+		controllerNeedChange = true
+	}
+	if preConfig.Spec.AntreaImage != curConfig.Spec.AntreaImage {
+		agentNeedChange = true
+		controllerNeedChange = true
+	}
+	return agentNeedChange, controllerNeedChange
+}
+
+func HasClusterNetworkConfigChange(preConfig, curConfig *configv1.Network) bool {
+	if preConfig == nil {
+		return true
+	}
+	if !stringSliceEqual(preConfig.Spec.ServiceNetwork, curConfig.Spec.ServiceNetwork) {
+		return true
+	}
+	var preCIDRs, curCIDRs []string
+	for _, clusterNet := range preConfig.Spec.ClusterNetwork {
+		preCIDRs = append(preCIDRs, clusterNet.CIDR)
+	}
+	for _, clusterNet := range curConfig.Spec.ClusterNetwork {
+		curCIDRs = append(curCIDRs, clusterNet.CIDR)
+	}
+	if !stringSliceEqual(preCIDRs, curCIDRs) {
+		return true
+	}
+	return false
+}
+
+func HasDefaultMTUChange(preConfig, curConfig *operatorv1.AntreaInstall) (bool, int, error) {
+
+	curAntreaAgentConfig := make(map[string]interface{})
+	err := yaml.Unmarshal([]byte(curConfig.Spec.AntreaAgentConfig), &curAntreaAgentConfig)
+	if err != nil {
+		return false, types.DefaultMTU, err
+	}
+	curDefaultMTU, ok := curAntreaAgentConfig[types.DefaultMTUOption]
+	if !ok {
+		return false, types.DefaultMTU, fmt.Errorf("%s option can not be empty", types.DefaultMTUOption)
+	}
+
+	if preConfig == nil {
+		return true, curDefaultMTU.(int), nil
+	}
+
+	preAntreaAgentConfig := make(map[string]interface{})
+	err = yaml.Unmarshal([]byte(preConfig.Spec.AntreaAgentConfig), &preAntreaAgentConfig)
+	if err != nil {
+		return false, types.DefaultMTU, err
+	}
+	preDefaultMTU, ok := preAntreaAgentConfig[types.DefaultMTUOption]
+	if !ok {
+		return false, types.DefaultMTU, fmt.Errorf("%s option can not be empty", types.DefaultMTUOption)
+	}
+
+	return preDefaultMTU != curDefaultMTU, curDefaultMTU.(int), nil
+}
+
+func BuildNetworkStatus(clusterConfig *configv1.Network, defaultMTU int) *configv1.NetworkStatus {
+	// Values extracted from spec are serviceNetwork and clusterNetworkCIDR.
+	// HostPrefix is ignored.
```

Comment:
Remove this comment as for antrea `HostPrefix` is useful

Reply:
Done.